rholang/src/main/scala/coop/rchain/rholang/interpreter/PrettyPrinter.scala

```diff
@@ -145,30 +154,30 @@ case class PrettyPrinter(freeShift: Int,
             })
         }
 
-        "for( " + bindsString + " ) { " + this
+        "for( " + bindsString + " ) {\n" + this
           .copy(boundShift = boundShift + totalFree)
-          .buildString(r.body) + " }"
+          .buildString(r.body, indent + 1) + "\n}"
 
       case b: Bundle =>
         BundleOps.showInstance.show(b) + "{ " + buildString(b.body) + " }"
 
       case n: New =>
-        "new " + buildVariables(n.bindCount) + " in { " + this
+        "new " + buildVariables(n.bindCount) + " in {\n" + this
           .copy(boundShift = boundShift + n.bindCount)
-          .buildString(n.p) + " }"
+          .buildString(n.p, indent + 1) + "\n}"
 
       case e: Expr =>
         buildString(e)
 
       case m: Match =>
-        "match " + buildString(m.target) + " { " +
+        "match " + buildString(m.target) + " {\n" +
           ("" /: m.cases.zipWithIndex) {
             case (string, (matchCase, i)) =>
               string + buildMatchCase(matchCase) + {
-                if (i != m.cases.length - 1) " ; "
+                if (i != m.cases.length - 1) " ;\n"
                 else ""
               }
-          } + " }"
+          }.split('\n').map("\t" * (indent + 1) + _).mkString("\n") + "\n}"
```

Comment:
please extract a `val Indent = '\t'` constant and use it all around

Reply:
Also, please make the indent two spaces instead of a tab.

Reply:
Also, don't split-map-mkstring indent elements in the body of the fold instead.

Reply:
I have use two spaces instead of a tab. 
And for `split-map-mkString`, it is needed. Because you should handle the indent in the view of newline rather than sub-rule of grammar, that will make the problem difficult to handle when component are display in the same line.